### DISCUSSION TITLE
How expensive is using S3 as a backend with SFTPGo?

### DISCUSSION THREAD

**Discussion by machty** [drakkan/sftpgo#1368] [Q&A]
I built my own version of SFTPGo a few years back using s3fuse, such that users log into the SFTP server and their home folder is essentially a subfolder within an s3fuse file system mount (in other words, their home folder resides within S3).

This has worked well for us but after a recent AWS cost audit, we found that this is actually costing us a lot of money in terms of S3 costs. In particular, running an SFTP server that mounts s3fuse to an s3 bucket internally causes tons of "$0.005 per 1,000 PUT, COPY, POST, or LIST requests", which ended up costing us about $20 per day, and it's not even that particularly active of an S3 server.

So I'm wondering, does anyone have any numbers on S3 costs when using S3 as a backend for sftpgo? (And would this number be any less for Google Storage or other competitors?)

**Comment by drakkan** [comment]
I've never used s3fs so I don't know how it works. If I understand your numbers correctly you should be making about 4 million requests a day. Maybe your SFTP clients make many more requests than you are aware of. 

You can try SFTPGo yourself and check your AWS bills, no one has ever complained about high costs before, at least that I know of

**Comment by machty** [comment]
> Maybe your SFTP clients make many more requests than you are aware of.

Yes, did some CloudTrail sleuthing and it turns out we have one client in particular hammering our SFTP server with way-too-aggressive polling scripts. With some tuning I think the S3 costs will come down to about $0.50/day from $20/day.
